python/llama-index-server/llama_index/server/api/utils/workflow.py

```
@@ -0,0 +1,115 @@
+        cls,
+        chat_id: str,
+        workflow: Workflow,
+    ) -> Context:
+        file_path = cls.get_storage_path(chat_id)
+        if not file_path.exists():
+            raise FileNotFoundError(f"No checkpoint found for chat_id: {chat_id}")
+        try:
+            with open(file_path, "r") as f:
+                ctx_data = json.load(f)
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Invalid checkpoint data for chat_id {chat_id}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)